overlord/snapstate/handlers.go

```diff
@@ -1690,15 +1620,42 @@ func (m *SnapManager) startSnapServices(t *state.Task, _ *tomb.Tomb) error {
 	perfTimings := timings.NewForTask(t)
 	defer perfTimings.Save(st)
 
-	_, snapst, err := snapSetupAndState(t)
+	snapsup, snapst, err := snapSetupAndState(t)
 	if err != nil {
 		return err
 	}
-
 	currentInfo, err := snapst.CurrentInfo()
 	if err != nil {
 		return err
 	}
+
+	// check if any previously disabled services are now no longer services and
+	// log messages about that
+	for _, svc := range snapst.LastActiveDisabledServices {
+		app, ok := currentInfo.Apps[svc]
+		if !ok {
+			logger.Noticef("previously disabled service %s no longer exists", svc)
+		} else if !app.IsService() {
+			logger.Noticef("previously disabled service %s is now an app and not a service", svc)
+		}
+	}
+
+	// get the services which should be disabled (not started),
+	// as well as the services which are not present in this revision, but were
+	// present and disabled in a previous one and as such should be kept inside
+	// snapst for persistent storage
+	svcsToSave, svcsToDisable, err := missingDisabledServices(snapst.LastActiveDisabledServices, currentInfo)
```

Comment:
I didn't remark this before but if one reads the comment it hints that maybe the return values of this helper should be

svcsToDisable, svcsToSave, err